### PR TITLE
Add training position assignments

### LINKED ISSUE #22: [BE] Training Department - I can define a position as being available for mentoring
### Description:
Not every position loaded into the database should be able to be mentored on. The training department should be able to define which positions they would like to permit mentoring on. A history of this should be recorded. 

### Accceptance Criteria / Requirements:
AC1: A position can be marked as having mentoring available

### PR DESCRIPTION
**Pull request by AxonC**
Closes #22.

- Adds the ability to mark & unmark positions as available for training
- Adds ability to query these positions

### CODE REVIEW

File: app/Modules/Position/PositionNotAssignedForTrainingException.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * Return the content that is put in the "extensions" part
+     * of the returned error.
+     *
+     * @return array
+     */
+    public function extensionsContent(): array
+    {
+        return [
+            'code' => 422,
```
Comment (AndyTWF) [inline app/Modules/Position/PositionNotAssignedForTrainingException.php]: This code should probably be 404.

File: database/migrations/2020_01_07_200042_create_training_position_assignments_table.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('training_position_assignments', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->unsignedBigInteger('position_id');
```
Comment (AndyTWF) [inline database/migrations/2020_01_07_200042_create_training_position_assignments_table.php:18]: Foreign key here?

**Review by AndyTWF — CHANGES_REQUESTED** [review]
Mostly just questions

File: database/migrations/2020_01_07_200042_create_training_position_assignments_table.php (hunk 2 above)
Comment (AxonC) [reply database/migrations/2020_01_07_200042_create_training_position_assignments_table.php:18]: added :)

File: app/Modules/Position/PositionNotAssignedForTrainingException.php (hunk 1 above)
Comment (AxonC) [reply app/Modules/Position/PositionNotAssignedForTrainingException.php]: good catch!

**Review by AndyTWF — APPROVED** [review]

**Pull request merged**